MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs

```diff
@@ -85,13 +85,16 @@ public ImageDocument ()
 				ImageView.Image = image;
 			}
 
-			Class [] classArray2 = { new Class ("ImageInfo") };
-			ok = pasteboard.CanReadObjectForClasses (classArray2, null);
-			if (ok) {
+			Class[] classArray2 = { new Class("ImageInfo") };
+			ok = pasteboard.CanReadObjectForClasses(classArray2, null);
+			if (ok)
+			{
 				// Read the image off of the pasteboard
-				NSObject [] objectsToPaste = pasteboard.ReadObjectsForClasses (classArray2, null);
-				ImageInfo info = (ImageInfo)objectsToPaste[0];
-
+				NSObject[] objectsToPaste = pasteboard.ReadObjectsForClasses(classArray2, null);
+				if (objectsToPaste.Length > 0)
+				{
+					ImageInfo info = (ImageInfo)objectsToPaste[0];
```

Comment:
I do not have an idea how the initial sample worked. Apple removes old samples, I will verify and make changes.